### DISCUSSION TITLE
limit_nonpersitant and limit_persitant always the same value

### DISCUSSION THREAD

**Discussion by tkuhn001** [tbnobody/OpenDTU#1988] [General]
I have a Hoymiles HMS-1600-4t and use Homeassitant and openDTU. Everything works great so far. I only have one problem:
In some threads you can read that you should not constantly change the limit with limit_persitant_relative or _absolute because otherwise the memory can break after a few years if you change the value many thousands of times. It is better to work with limit_nonpersitant_relative or _absolute.
If I now set my limit_persitant_relative to 50 in openDTU, the performance on all panels is throttled to 50%. If I query limit_nonpersitant_relative, the value is also 50. If I now set limit_nonpersitant_relative to 60, the power on all panels is set to 60%. However, if I now query the value of limit_persitant_relative, this is also 60 although it should actually be 50. It is the same the other way round. Is this a bug?
Thx for reading

SDK-Version	v4.4.6-dirty
Konfigurationsversion	0.1.28
Firmwareversion / git Hash	[v24.5.6](https://github.com/tbnobody/OpenDTU/releases/tag/v24.5.6)
PIO Umgebung	generic_esp32s3_usb

**Comment by MetaChuh** [comment]
@tkuhn001 

please do not worry, you've done everything correctly.
currently our reverse engineered dtus are not able to distinguish between persistent and non persistent readback values.

glad to read, no thx needed for that & greetings,
metachuh and our whole solar community

**Comment by jubie25** [comment]
Got a HM-800-2T connected via OpenDTU. Firmware of the Hoymiles is 1.0.10 with date code 2021-11-01 08:42:00. And I can confirm that **non-persist values are stored in persist memory, too**.
Test: 
1. Home Assistant, set both values to 800W, then set non-persist value to 47W. Observe power producion in OpenDTU: fits to the temporarily set limit. Limit is also shown in OpenDTU.
2. cut Hoymiles from power (AC and both panels). Wait for LED stop blinking and add extra 30s. 
3. connect all again
4. OpenDTU gets it's first values again from Hoymiles, showing 47W limit. Producing power fits to it, again.
5. Home Assistant shows 47W again als limit, for persistant ans non persistant storage.

Test 2: resetting Hoymiles via restart command in OpenDTU: this resets both values to default (800W) and is NOT the same as simply disconnect from all power sources!

So it's clear that those non-persistant limits are stored persistant anyway!

Now is there a way to set

**Comment by jubie25** [comment]
O M G...
I did my tests with the WRONG Hoymiles unit. Taking the right one confirms that non  persist values are resetted with a power cycle, so they are obviously stored in RAM (only)!
Sorry for nothing!!
